source/material_model/grain_size.cc

```diff
@@ -275,11 +293,11 @@ namespace aspect
                                                        / (geometric_constant[phase_index] * grain_boundary_energy[phase_index]);
               grain_size_reduction = grain_size_reduction_rate * grain_growth_timestep;
             }
-          else if (grain_size_evolution_formulation == Formulation::pinned_grain_damage)  
+          else if (grain_size_evolution_formulation == Formulation::pinned_grain_damage)
             {
               // pinned_grain_damage: Mulyukova and Bercovici (2018) Collapse of passive margins by lithospheric damage and plunging grain size. Earth and Planetary Science Letters, 484, 341-352.
               const double stress = 2.0 * second_strain_rate_invariant * current_viscosity;
-              const double grain_size_reduction_rate = 2.0 * stress * boundary_area_change_work_fraction[phase_index] * second_strain_rate_invariant * pow(grain_size,2)
+              const double grain_size_reduction_rate = 2.0 * stress * compute_partitioning_fraction(temperature) * second_strain_rate_invariant * pow(grain_size,2)
```

Comment:
This is the same place I change in #9. So @alarshi it might be the easiest if you include all the changes I already have in #9 and then I can close that PR.

Reply:
Got it, thank you! I added your changes in the latest push.